pkg/apis/workspaces/v1alpha2/component_image_dockerfile.go

```diff
@@ -66,3 +62,11 @@ type Dockerfile struct {
 	// +optional
 	RootRequired bool `json:"rootRequired,omitempty"`
 }
+
+type DockerfileGitProjectSource struct {
+	GitProjectSource `json:",inline"`
```

Comment:
Should it really be GitProjectSource? Like why dockerfileGitSource needs a `name`, `attributes`, `clonePath`?

Reply:
I think the main concern was reusing the struct from projects/starterprojects

>  Like why dockerfileGitSource needs a name, attributes, clonePath?

I'm not sure what you mean -- with this change, a component would look something like
```yaml
components:
  - name: my-component
    image:
      git:
        gitLocation: "build/Dockerfile"
        checkoutFrom:
          remote: amisevsk
          revision: devel-branch
        remotes:
          origin: https://github.com/devfile/api
          amisevsk: https://github.com/amisevsk/api
```
(with checkoutFrom optional when there's only one remote)

Reply:
that's not related to your PR, but with 
```go
type DockerfileGitProjectSource struct {
	GitProjectSource `json:",inline"`
```

format may is:
```
components:
  - name: my-component
    image:
      git:
        name: project-name
        clonePath: "does not matter but it's here"
        gitLocation: "build/Dockerfile"
        checkoutFrom:
          remote: amisevsk
          revision: devel-branch
        remotes:
          origin: https://github.com/devfile/api
          amisevsk: https://github.com/amisevsk/api
```

I suppose we need a dedicated GitDockerfileSource, instead of `GitProjectSource`